lib/main.coffee

```diff
@@ -9,71 +8,49 @@ module.exports =
 
   provideLinter: ->
     provider =
+      name: 'foodcritic'
       grammarScopes: ['source.ruby.chef']
       scope: 'file'
       lintOnFly: false
       lint: (textEditor) =>
-        return new Promise (resolve, reject) =>
-          currentFilePath = textEditor.getPath()
-          fileDir = pathModule.dirname(currentFilePath)
-          cwd = pathModule.dirname(helpers.findFile(fileDir, 'metadata.rb'))
-          if cwd == '.'
-            resolve []
-          currentFileRelPath = pathModule.relative(cwd, currentFilePath)
-          # change filePath to be relative to @cwd
-          filePath = currentFileRelPath.replace(/\\/g, '/').replace(/^\//, '')
-          regex = "(?<text>.+:\\s.+):\\s+(./)?(?<filePath>#{filePath}|[\\w+\\._]+):(?<line>\\d+)"
-          fOutput = ''
-          if @extraArgs
-            args = @extraArgs.split(' ').concat ['.']
-          else
-            args = ['.']
-          process = new BufferedProcess
-            command: @executablePath
-            args: args
-            options:
-              cwd: cwd
-            stdout: (data) ->
-              fOutput += data
-            exit: (code) ->
-              console.log("out: #{fOutput}")
-              return resolve [] unless code is 0
-              return resolve [] unless fOutput?
-              messages = []
-              @MessageRegexp ?= XRegExp regex, ''
-              XRegExp.forEach fOutput, @MessageRegexp, (match, i) ->
-                match.line ?= 0
-                normFilePath = pathModule.normalize(match.filePath)
-                if normFilePath == currentFileRelPath
-                  msg = {
-                    type: 'Error',
-                    text: match.text,
-                    filePath: currentFilePath,
-                    range: helpers.rangeFromLineNumber(textEditor, match.line - 1)
-                  }
-                  messages.push msg if msg.range?
-              , this
-              resolve messages
-
-          process.onWillThrowError ({error,handle}) ->
-            atom.notifications.addError "Failed to run #{@executablePath}",
-              detail: "#{error.message}"
-              dismissable: true
-            handle()
-            resolve []
+        currentFilePath = textEditor.getPath()
+        fileDir = pathModule.dirname(currentFilePath)
+        cwd = pathModule.dirname(helpers.find(fileDir, 'metadata.rb'))
+        if cwd is '.'
+          atom.notifications.addWarning('[foodcritic] No metadata.rb found, not linting!', {dismissable: true})
+          return Promise.resolve([])
+        if @extraArgs
+          args = @extraArgs.split(' ').concat ['.']
+        else
+          args = ['.']
+        regex = @MessageRegexp
+        helpers.exec(@executablePath, args, {cwd: cwd}).then (output) ->
+          return [] unless output?
+          messages = []
+          while((match = regex.exec(output)) isnt null)
+            match.line = 1 if not match[2]? or match[2] < 1
```

Comment:
Although an invalid line would be `undefined`, not `null`, so you are right that it isn't quite correct.